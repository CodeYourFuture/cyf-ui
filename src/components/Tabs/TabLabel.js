import React from "react";
import PropTypes from "prop-types";
import styled, { css } from "styled-components";

const Li = styled.li`
  display: inline-block;
  list-style: none;
  margin-bottom: -1px;
  margin-right: 0.5rem;
  background-color: lightgrey;
  color: #074a95;
  font-weight: 700;
  padding: 1rem 1.5rem;
  cursor: pointer;

  ${({ active }) =>
    active &&
    css`
      background-color: white;
      border: solid rgb(246, 245, 247);
      border-width: 1px 1px 0 1px;
    `}
`;

export const TabLabel = ({ clickTabItem, active, label }) => {
  return (
    <Li active={active} onClick={() => clickTabItem(label)} aria-hidden="true">
      {label}
    </Li>
  );
};

TabLabel.propTypes = {
  active: PropTypes.bool.isRequired,
  label: PropTypes.string.isRequired,
  clickTabItem: PropTypes.func.isRequired,
};
