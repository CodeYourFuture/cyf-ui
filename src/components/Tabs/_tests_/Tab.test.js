import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { Tab } from "..";
import { TabLabel } from "../TabLabel";

describe("<Tab/>", () => {
  test("renders without crushing", () => {
    render(<Tab label="Steps" />);
  });
  test("displays the given label name", () => {
    const label = "NewCalls";
    const { getByText } = render(
      <Tab label={label}>
        <p>This is Tab 0 NewCalls!</p>
      </Tab>
    );
    expect(getByText(/This is Tab 0 NewCalls!/i)).not.toBeNull();
  });
});

describe("<TabLabel/>", () => {
  test("renders without crushing", () => {
    const activeTab = false;
    const label = "";
    const clickTabItem = jest.fn(() => null);
    render(
      <TabLabel active={!activeTab} label={label} clickTabItem={clickTabItem} />
    );
    expect(clickTabItem).toMatchSnapshot();
  });
  test("clickTab", () => {
    const mockOnClick = jest.fn();
    const activeTab = false;
    const testLabel = "Workshops";

    const { getByTestId } = render(
      <TabLabel
        active={!activeTab}
        label={testLabel}
        clickTabItem={mockOnClick}
      />
    );
    const tabList = getByTestId("labels");
    fireEvent.click(tabList);
    expect(mockOnClick).toHaveBeenCalledWith(testLabel);
  });
  test("clickTab with mocked function", () => {
    const activeTab = false;
    const label = "Steps";
    const click = jest.fn(() => null);
    const { getByText } = render(
      <TabLabel active={!activeTab} label={label} clickTabItem={click} />
    );
    const tabList = getByText(label);
    fireEvent.click(tabList);
    expect(click).toHaveBeenCalled();
  });
});
