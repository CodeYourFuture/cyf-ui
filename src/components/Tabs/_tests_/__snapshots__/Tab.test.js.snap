// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabLabel/> renders without crushing 1`] = `[MockFunction]`;
